src/realm-settings/key-providers/aes-generated/AESGeneratedForm.tsx

```diff
@@ -0,0 +1,363 @@
+              }
+              onChange={(value) => {
+                onChange([value + ""]);
+              }}
+            />
+          )}
+        />
+      </FormGroup>
+      <FormGroup
+        label={t("realm-settings:active")}
+        fieldId="kc-active"
+        labelIcon={
+          <HelpItem
+            helpText="realm-settings-help:active"
+            forLabel={t("active")}
+            forID="kc-active"
+          />
+        }
+      >
+        <Controller
+          name="config.active"
+          control={form.control}
+          defaultValue={["true"]}
+          render={({ onChange, value }) => {
+            return (
+              <Switch
+                id="kc-active"
+                label={t("common:on")}
+                labelOff={t("common:off")}
+                isChecked={value[0] === "true"}
+                data-testid={
+                  value[0] === "true"
+                    ? "internationalization-enabled"
+                    : "internationalization-disabled"
+                }
+                onChange={(value) => {
+                  onChange([value + ""]);
+                }}
+              />
+            );
+          }}
+        />
+      </FormGroup>
+      <FormGroup
+        label={t("realm-settings:AESKeySize")}
+        fieldId="kc-aes-keysize"
+        labelIcon={
+          <HelpItem
+            helpText="realm-settings-help:AESKeySize"
+            forLabel={t("AESKeySize")}
+            forID="kc-aes-key-size"
+          />
+        }
+      >
+        <Controller
+          name="config.secretSize"
+          control={form.control}
+          defaultValue={["16"]}
+          render={({ onChange, value }) => (
+            <Select
+              toggleId="kc-aes-keysize"
+              onToggle={() => setIsKeySizeDropdownOpen(!isKeySizeDropdownOpen)}
+              onSelect={(_, value) => {
+                onChange([value.toString()]);
+                setIsKeySizeDropdownOpen(false);
+              }}
+              selections={[value + ""]}
+              isOpen={isKeySizeDropdownOpen}
+              variant={SelectVariant.single}
+              aria-label={t("aesKeySize")}
+              data-testid="select-secret-size"
+            >
+              {AESSecretSizeOptions?.map((item, idx) => (
+                <SelectOption
+                  selected={item === value}
+                  key={`secret-size-option-${idx}`}
+                  value={item}
+                />
+              ))}
+            </Select>
+          )}
+        />
+      </FormGroup>
+      <ActionGroup className="kc-AESform-buttons">
+        <Button
+          className="kc-AESform-save-button"
+          data-testid="add-provider-button"
+          variant="primary"
+          type="submit"
+        >
+          {t("common:save")}
+        </Button>
+        <Button
+          className="kc-AESform-cancel-button"
+          onClick={!editMode ? () => handleModalToggle?.() : () => {}}
```

Comment:
Instead of falling back to an empty function, it's also possible to pass in `undefined`.
```suggestion
          onClick={(!editMode && handleModalToggle) || undefined}
```